### ISSUE TITLE
Coding Style Conventions

### ISSUE THREAD

**Issue by martinjlowm** [elysium-project/core#5]
First and foremost, it's nice to see the core finally hit the repository. Great job!

Secondly, I have already had a brief look at some of the source code and there are some inconsistencies in the coding style which I think we should clear up.

- Much of the code has 4 space indentation, but 8 spaces are also used some places.
- Some `if`-blocks have curly braces around them, others do not, i.e. single statements. This will probably lead to a religious discussion, but remember, we want to avoid bugs. Leaving curly braces out in the single statement situation, _will_ produce a bug at some point (I look at you BattleGround.cpp:199!).
- Should there be a limit to the number of columns? In my opinion, yes, 80 characters limit is appropriate.
- Scope specifiers (private, public, protected) in classes are sometimes indented and sometimes not. I recommend half the normal indentation, thus 2 spaces.
- Some functions that are _not_ inline are implemented in header files.
- Objects' fields sometimes use the `m_field_name` convention and sometimes just `_field_name`.
- The use of namespaces seems random. Sometimes they are related to directory names, sometimes not.
- The include path fed to the compiler is recursive. It's difficult to find header files based on the `#include` macros. I recommend having `${CMAKE_CURRENT_SOURCE_DIR}` as the base include directory and add the relative path to the `#include` macros.

Another thing is C++ header files that have the `.h` extension. While I know this is common in many projects, it can still be confusing (I think so at least) because the extension refers to a C header file. 
Changing all headers from `.h` to `.hpp` or `.hh` might be a bit much, instead, I recommend adding `/* -*- c++ -*- */` to the start of the files, such that Emacs (which I use) detect the correct syntax.

**Comment by illutian** [comment]
I noticed that the SQL tables use " ` ", now I haven't delved deep into SQL, but I have never once used the " ` " in my queries. But I have seen in document files (including upload templates) where people will use " ` " instead of " ' ". And it'll either get flagged by grammar/spelling checks (documents) or error out an upload (xml files, etc).

So, in SQL, is the " ` " normal? Or should it be changed to " ' "?

**Comment by martinjlowm** [comment]
By tilde key apostrophe you mean \`, then yes, that's normal. Either works.

**Comment by StadenElysium** [comment]
Thanks for your suggestions. Style conventions will be added shortly (probably the weekend).

**Comment by illutian** [comment]
@martinjlowm yes, that...Git decided I wanted the entire block of text to be highlighted in grey (can tell how much I use this place! :P [NO, I DO NOT WANT TO DO A PACKAGE, GIT!]).

In that case, I would say the standard for syntax should also be established. This was mentioned for the C++ part, but not for the SQL; I'm simply saying it should be applied globally, to the entire project.

**Comment by ElyMage** [comment]
Should we just follow this?

https://github.com/cmangos/issues/wiki/Coding-Standards
